lua/telescope/builtin/init.lua

```diff
@@ -21,58 +30,248 @@ end
 
 local builtin = {}
 
---- Live grep means grep as you type.
+
+--------------------------------------------------
+--
+-- File-related Pickers
+--
+--------------------------------------------------
+--- Search for a string in your current working directory and get results live as you type (respecting .gitignore)
+---@param opts table: options to pass to the picker
+---@field grep_open_files boolean: restrict search to open files only, mutually exclusive with `search_dirs`
+---@field search_dirs table: directory/directories to search in, mutually exclusive with `grep_open_files`
 builtin.live_grep = require('telescope.builtin.files').live_grep
 
+--- Searches for the string under your cursor in your current working directory
+---@param opts table: options to pass to the picker
+---@field search string: the query to search
+---@field search_dirs table: directory/directories to search in
 builtin.grep_string = require('telescope.builtin.files').grep_string
+
+--- Lists files in your current working directory, respects .gitignore
+---@param opts table: options to pass to the picker
+---@field find_command table: command line arguments for `find_files` to use for the search, overrides default config
+--TODO @field follow boolean:
+---@field hidden boolean: determines whether to show hidden files or not
+---@field search_dirs table: directory/directories to search in
 builtin.find_files = require('telescope.builtin.files').find_files
+
 builtin.fd = builtin.find_files
+
+--- Lists files and folders in your current working directory, open files, navigate your filesystem, and create new
+--- files and folders
+--- - Default keymaps:
+---   - <cr> type: opens the currently selected file, or navigates to the currently selected directory
+---   - <C-e> type: creates new file in current directory, creates new directory if the name contains a trailing '/'
```

Comment:
we should also probably tell if it's possible to do `path/to/file.txt` and it will create nested directories. Some people don't know if this is possible, it's useful for Java where you'd have `a/long/nested/dir/for/whatever/Reason.java`